src/net_processing.cpp

```diff
@@ -2067,10 +2103,13 @@ bool static ProcessMessage(CNode* pfrom, const std::string& strCommand, CDataStr
 
         // Potentially mark this peer as a preferred download peer.
         {
-        LOCK(cs_main);
-        UpdatePreferredDownload(pfrom, State(pfrom->GetId()));
+            LOCK(cs_main);
+            UpdatePreferredDownload(pfrom, State(pfrom->GetId()));
         }
 
+        // Check the peer's version and compare it to us
+        CheckForUpdates(pfrom->addr.ToString(), pfrom->strSubVer);
+
```

Comment:
I'm not sure about this here.  I could be wrong, but looking at the whole logic flow, it looks like we're going to check each one individually but start setting things after we have the first few (see later)

Reply:
Yep.